spec/lending.md

```diff
@@ -1,13 +1,39 @@
+
+Each reserve's **Interest Rate**, the percentage that borrowers must pay on their loans, is the primary tool used to curve the current utilization rate towards the optimal one. Increasing the interest rate is expected to decrease the utilization rate and vice versa. 
+
+The interest rate is fixed per borrower at the time of the loan. Therefore we are likely to observe asymmetrical sensitivity to interest changes. This means that decreasing the interest is likely to result in a relatively quick increase in the utilization rate. However the contrary, increasing the interest rate, doesn't affect current borrowers (only potential new ones) and thus we would expect a much slower decrease in utilization rate. This is something that the controlling algorithm must take into consideration.
+
+Lastly, points of crisis are when lenders try to withdraw more money than is currently available or when borrowers try to borrow more money than is available. The interest rates are used to avoid crisis by trying to achieve a safe utilization rate, however, in the event this happens the transaction will simply be denied. This means that lenders take on the risk of a) the borrowers defaulting and b) potential illiquidity of their tokens.  
+
+## Advanced Features
+
+The basics of the protocol can be furthered by a set of possible advanced features:
+
+- **Lending epochs:** To avoid hight volatility in the reserve pool, tokens that are lent could only be withdrawn at prescribed intervals or epochs. This would break the withdraw call into two calls: First, an intention to withdraw which would then set a time in the future with which the tokens become unlocked; Second, the actual withdrawal of tokens so long as the call is made after the tokens are unlocked. 
+
+- **Deposit Rate Limiting:** Another feature to combat high volatility is to set a ratio with which funds can be deposited or withdrawn. As an example, every epoch only 10% of the total liquidity in the reserve can be exchanged. This could also be applied to borrowing.
+
+- **Variable Borrowing Rate:** Currently the protocol charges a fixed interest rate to borrowers because less overhead is required and it is expected that most
+borrowers prefer a fixed interest rate. If a reserve were to keep track of the interest rate changes over time then it could be easier to calculate accrued interest over the loan period and thus use variable rates. Variable rates offer better sensitivity to changes in utilization rate. We could further extend this to allow for both rate types and the ability for the borrower to swap in between. 
```

Comment:
We might want to discuss whether we want to go first with fixed interest rates or variable ones. I see each having benefits and drawbacks. Sure at a later point we can integrate both have `swap rate` calls

Reply:
Think its fine to not go into so much detail here. If we have extra time in the hackathon we can work on the details

Reply:
Sure I've just gone with the simple linear model that the compound protocol uses. I agree that it suffices for now

Reply:
one thing im reading into now is if we should charge the interest on the opening of the margin loan or after. Do you have thoughts?

Reply:
I actually thought about this, you could transfer say a portion of the collateral in some intermediary account and that way you could effectively repay the interest slowly over time rather than in one chunk. The annoying thing is that you don't know in advance how long the loan is going to be.